### PR TITLE
Changed useLayoutEffect to useIsomorphicLayoutEffect

### PR DESCRIPTION
**Pull request by schultzp2020**
https://github.com/patternfly/patternfly-react/issues/5953

**Additional issues**:
None

### CODE REVIEW

File: packages/react-core/src/helpers/Popper/thirdparty/react-popper/usePopper.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ const fromEntries = (entries: [string, any][]) =>
 /**
  * Small wrapper around `useLayoutEffect` to get rid of the warning on SSR envs
  */
-const useIsomorphicLayoutEffect =
+export const useIsomorphicLayoutEffect =
   canUseDOM
     ? React.useLayoutEffect
     : React.useEffect;
```
Comment (redallen) [inline packages/react-core/src/helpers/Popper/thirdparty/react-popper/usePopper.ts:26]: This is a strange place for this function to live. Can you move it to a new `src/helpers/useIsomorphicLayout.ts` file that gets reexported from `src/helpers/index.ts` instead?

File: packages/react-core/src/components/Modal/ModalBoxTitle.tsx [hunk 2]
```diff
@@ -9,6 +9,7 @@ import ExclamationCircleIcon from '@patternfly/react-icons/dist/js/icons/exclama
 import ExclamationTriangleIcon from '@patternfly/react-icons/dist/js/icons/exclamation-triangle-icon';
 import InfoCircleIcon from '@patternfly/react-icons/dist/js/icons/info-circle-icon';
 import BellIcon from '@patternfly/react-icons/dist/js/icons/bell-icon';
+import { useIsomorphicLayoutEffect } from '../../helpers/Popper/thirdparty/react-popper/usePopper';
```
Comment (redallen) [inline packages/react-core/src/components/Modal/ModalBoxTitle.tsx:12]: These imports can look like `import { useIsomorphicLayoutEffect } from '../../helpers';`

**Review by redallen — CHANGES_REQUESTED** [review]
Shuffle around this function and this PR looks good! I'll write a linting rule to catch `useLayoutEffect` once it's in.

**Comment by schultzp2020** [comment]
Do you know where the `Circular dependency` issue is coming from?

**Comment by redallen** [comment]
> dist/esm/helpers/index.js -> dist/esm/helpers/Popper/Popper.js -> dist/esm/helpers/Popper/thirdparty/react-popper/usePopper.js -> dist/esm/helpers/index.js

Solution: have usePopper.js import from helpers/useIsomorphicLayout.js instead

**Comment by schultzp2020** [comment]
@redallen is this good?

**Review by redallen — DISMISSED** [review]
Thanks for the contribution! We'll need a followup linting rule to disallow `useLayoutEffect` (#5970).

**Comment by schultzp2020** [comment]
Is this `.eslintrc.json` rule fine?

``` json
"no-restricted-imports": [
    "error",
    {
      "name": "react",
      "importNames": ["useLayoutEffect"],
      "message": "`useLayoutEffect` causes a warning in SSR. Use `useIsomorphicLayoutEffect` from `patternfly-react/packages/react-core/src/helpers`"
    }
  ],
```

**Comment by redallen** [comment]
@schultzp2020 Yep that looks good!

**Comment by schultzp2020** [comment]
@redallen Should the rule change since there's an error on each `import * from 'react'`.

**Comment by schultzp2020** [comment]
@redallen I cant get the rule to turn on. Can you look at the imports?

**Comment by redallen** [comment]
@schultzp2020 It looks like it's [working great in CI!](https://app.circleci.com/pipelines/github/patternfly/patternfly-react/12522/workflows/88c6a1dd-b82e-4a74-b38c-4ab48d0858a0/jobs/60569)

image

**Comment by schultzp2020** [comment]
Weird, on my on it says it cant find the rule

File: packages/react-topology/src/components/popper/Portal.tsx [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import * as React from 'react';
 import * as ReactDOM from 'react-dom';
+import { useIsomorphicLayoutEffect } from '../../../../react-core/src/helpers';
```
Comment (redallen) [inline packages/react-topology/src/components/popper/Portal.tsx:3]: Change to `from '@patternfly/react-core/dist/esm/helpers';`

Comment (schultzp2020) [reply packages/react-topology/src/components/popper/Portal.tsx]: image

Comment (redallen) [reply packages/react-topology/src/components/popper/Portal.tsx]: Ah my mistake. Try just `from '@patternfly/react-core';`.

**Comment by schultzp2020** [comment]
Let me know if you want me to change anything else.

**Review by redallen — APPROVED** [review]
LGTM! Thanks for adding the linting rule.

**Comment by schultzp2020** [comment]
@redallen Who is the other reviewer?

**Review by dlabrecq — APPROVED** [review]

**Pull request merged**